src/main/java/org/minions/devfund/reinaldo/battleship/Ship.java

```diff
@@ -0,0 +1,215 @@
+     *
+     * @param row        the row.
+     * @param column     the column.
+     * @param horizontal the orientation to ship.
+     * @param ocean      {@link Ocean}.
+     * @return true if is possible to place a ship in the position.
+     */
+    boolean okToPlaceShipAt(final int row, final int column, final boolean horizontal, final Ocean ocean) {
+        return horizontal ? column + length - 1 < Ocean.SHIPLENGTH
+                && isHorizontalOrVerticalLineOccupied(row, column, column, horizontal, ocean)
+                : row + length - 1 < Ocean.SHIPLENGTH
+                && isHorizontalOrVerticalLineOccupied(row, column, row, horizontal, ocean);
+    }
+
+    /**
+     * Verifies if horizontal or vertical line is occupied in the ocean.
+     *
+     * @param row        int ro position.
+     * @param column     int column position.
+     * @param iterator   int iterate.
+     * @param horizontal the orientation to ship.
+     * @param ocean      {@link Ocean}.
+     * @return true if line is occupied.
+     */
+    private boolean isHorizontalOrVerticalLineOccupied(final int row, final int column, final int iterator,
+                                                       final boolean horizontal, final Ocean ocean) {
+        for (int j = iterator; j < iterator + length; j++) {
+            final boolean occupiedHorizontal = ocean.isOccupied(row, j) || ocean.isBorderOccupied(row, j);
+            final boolean occupiedVertital = ocean.isOccupied(j, column) || ocean.isBorderOccupied(j, column);
+            final boolean isOccupied = horizontal ? occupiedHorizontal : occupiedVertital;
+            if (isOccupied) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Places a ship in a position.
+     *
+     * @param row        row for the ship.
+     * @param column     column for the ship.
+     * @param horizontal orientation of the ship.
+     * @param ocean      {@link Ocean}.
+     */
+    void placeShipAt(final int row, final int column, final boolean horizontal, final Ocean ocean) {
+        bowColumn = column;
+        bowRow = row;
+        setHorizontal(horizontal);
+        if (horizontal) {
+            for (int j = column; j < column + length; j++) {
+                ocean.getShipArray()[row][j] = this;
+            }
+        }
+        if (!horizontal) {
+            for (int i = row; i < row + length; i++) {
+                ocean.getShipArray()[i][column] = this;
+            }
+        }
+    }
+
+    /**
+     * Shots at a certain position.
+     *
+     * @param row    int the row position.
+     * @param column int the column position.
+     * @return true if a ship was shot.
+     */
+    boolean shootAt(final int row, final int column) {
+        if (isSunk()) {
+            return false;
+        }
+        boolean isHittable = horizontal ? row == bowRow && column >= bowColumn && column < bowColumn + length
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1067)